jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt

```diff
@@ -43,6 +43,7 @@ class DeployServerlessApplicationDialog(
     private val s3Client: S3Client = project.awsClient()
     private val cloudFormationClient: CloudFormationClient = project.awsClient()
     private val templateParameters = CloudFormationTemplate.parse(project, templateFile).parameters().toList()
+    private lateinit var stackNameToStackId: Map<String, String>
```

Comment:
Why not just store this mapping as a data class in the selector itself?

Reply:
I think selector shouldn't store this because this mapping is specific to stacks and the selector is a general purpose combo box storing not only stacks but also s3Buckets and iamRoles etc.

Reply:
The selector just stores data, you can store anything in a combo box and just stick a custom renderer on it.

Reply:
This is how we do IAM roles as well. Its a ResourceSelector<IamRole>

Reply:
Done, using ResourceSelector<Stack> now like ResourceSelector<IamRole>.